### ISSUE TITLE
Hundreds of localhost:8080 connections?

### ISSUE THREAD

**Issue by ChubbyNinja** [ratchetphp/Ratchet#822]
I'm trying to diagnose a server issue we experienced this morning, is it normal to have hundreds of localhost connections?
Or is there a bug in the way I'm using this?

For reference, we only have 5-6 users on the platform currently.

```
tcp        0      0 127.0.0.1:59388         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59436         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59432         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59370         ESTABLISHED
tcp        0      0 127.0.0.1:59284         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59204         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59590         ESTABLISHED
tcp        0      0 127.0.0.1:59238         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59314         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59414         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59240         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59608         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59332         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59356         ESTABLISHED
tcp        0      0 127.0.0.1:59354         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59376         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59400         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59410         ESTABLISHED
tcp        0      0 127.0.0.1:59598         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59260         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59202         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59220         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59230         ESTABLISHED
tcp        0      0 127.0.0.1:59462         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59176         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59214         ESTABLISHED
tcp        0      0 127.0.0.1:59202         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59166         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59256         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59612         ESTABLISHED
tcp        0      0 127.0.0.1:59296         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59266         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59326         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59434         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59346         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59240         ESTABLISHED
tcp        0      0 127.0.0.1:59258         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59588         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59430         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59190         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59606         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59264         ESTABLISHED
tcp        0      0 127.0.0.1:59402         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59478         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59604         ESTABLISHED
tcp        0      0 127.0.0.1:59384         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59220         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59424         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59272         ESTABLISHED
tcp        0      0 127.0.0.1:59404         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59258         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59318         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59374         ESTABLISHED
tcp        0      0 127.0.0.1:59210         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59378         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59368         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59442         ESTABLISHED
tcp        0      0 127.0.0.1:59172         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59318         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59390         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59304         ESTABLISHED
tcp        0      0 127.0.0.1:59548         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59244         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59294         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59198         ESTABLISHED
tcp        0      0 127.0.0.1:59184         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59590         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59300         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59364         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59296         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59392         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59416         ESTABLISHED
tcp        0      0 127.0.0.1:59226         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59160         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59204         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59236         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59492         ESTABLISHED
tcp        0      0 127.0.0.1:59206         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59358         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59346         ESTABLISHED
tcp        0      0 127.0.0.1:59456         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59406         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59320         ESTABLISHED
tcp        0      0 127.0.0.1:59302         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59596         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59270         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59610         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59186         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59594         ESTABLISHED
tcp        0      0 127.0.0.1:59362         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59272         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59254         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59352         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59376         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59216         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59262         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59434         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59622         ESTABLISHED
tcp        0      0 127.0.0.1:59410         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59420         ESTABLISHED
tcp        0      0 127.0.0.1:59282         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59426         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59330         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59446         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59174         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59302         ESTABLISHED
tcp        0      0 127.0.0.1:59420         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59448         ESTABLISHED
tcp        0      0 127.0.0.1:59438         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59452         ESTABLISHED
tcp        0      0 127.0.0.1:59264         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59266         ESTABLISHED
tcp        0      0 127.0.0.1:59156         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59274         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59166         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59610         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59390         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59354         ESTABLISHED
tcp        0      0 127.0.0.1:59214         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59254         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59626         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59162         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59232         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59172         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59372         ESTABLISHED
tcp        0      0 127.0.0.1:59242         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59330         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59396         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59444         ESTABLISHED
tcp        0      0 127.0.0.1:59436         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59468         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59300         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59184         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59364         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59328         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59332         ESTABLISHED
tcp        0      0 127.0.0.1:59478         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59370         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59418         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59210         ESTABLISHED
tcp        0      0 127.0.0.1:59294         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59146         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59208         ESTABLISHED
tcp        0      0 127.0.0.1:59604         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59270         ESTABLISHED
tcp        0      0 127.0.0.1:59452         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59310         ESTABLISHED
tcp        0      0 127.0.0.1:59368         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59212         ESTABLISHED
tcp        0      0 127.0.0.1:59208         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59328         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59198         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59310         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59360         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59454         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59388         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59402         ESTABLISHED
tcp        0      0 127.0.0.1:59396         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59588         ESTABLISHED
tcp        0      0 127.0.0.1:59228         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59312         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59586         ESTABLISHED
tcp        0      0 127.0.0.1:59304         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59366         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59404         ESTABLISHED
tcp        0      0 127.0.0.1:59428         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59366         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59602         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59242         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59180         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59348         ESTABLISHED
tcp        0      0 127.0.0.1:59382         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59600         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59132         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59194         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59282         ESTABLISHED
tcp        0      0 127.0.0.1:59454         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59386         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59260         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59406         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59456         ESTABLISHED
tcp        0      0 127.0.0.1:59586         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59136         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59194         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59148         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59628         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59596         ESTABLISHED
tcp        0      0 127.0.0.1:59646         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59538         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59286         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59414         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59400         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59274         ESTABLISHED
tcp        0      0 127.0.0.1:59446         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59314         ESTABLISHED
tcp        0      0 127.0.0.1:59298         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59218         ESTABLISHED
tcp        0      0 127.0.0.1:59316         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59316         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59226         ESTABLISHED
tcp        0      0 127.0.0.1:59232         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59178         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59386         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59548         ESTABLISHED
tcp        0      0 127.0.0.1:59158         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59622         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59416         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59356         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59444         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59492         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59144         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59180         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59606         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59160         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59288         ESTABLISHED
tcp        0      0 127.0.0.1:59592         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59212         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59462         ESTABLISHED
tcp        0      0 127.0.0.1:59380         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59612         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59320         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59394         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59350         ESTABLISHED
tcp        0      0 127.0.0.1:59408         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59608         ESTABLISHED
tcp        0      0 127.0.0.1:59244         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59146         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59438         ESTABLISHED
tcp        0      0 127.0.0.1:59378         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59428         ESTABLISHED
tcp        0      0 127.0.0.1:59458         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59374         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59150         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59432         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59538         ESTABLISHED
tcp        0      0 127.0.0.1:59392         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59190         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59178         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59646         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59424         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59238         ESTABLISHED
tcp        0      0 127.0.0.1:59148         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59286         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59216         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59186         ESTABLISHED
tcp        0      0 127.0.0.1:59348         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59448         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59408         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59156         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59312         ESTABLISHED
tcp        0      0 127.0.0.1:59350         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59132         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59170         ESTABLISHED
tcp        0      0 127.0.0.1:59176         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59592         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59384         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59572         ESTABLISHED
tcp        0      0 127.0.0.1:59288         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59372         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59628         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59218         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59158         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59430         ESTABLISHED
tcp        0      0 127.0.0.1:59230         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59352         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59602         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59228         ESTABLISHED
tcp        0      0 127.0.0.1:59256         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59144         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59168         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59150         ESTABLISHED
tcp        0      0 127.0.0.1:59262         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59174         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59170         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59284         ESTABLISHED
tcp        0      0 127.0.0.1:59394         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59382         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59600         ESTABLISHED
tcp        0      0 127.0.0.1:59426         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59162         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59594         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59362         ESTABLISHED
tcp        0      0 127.0.0.1:59168         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59136         ESTABLISHED
tcp        0      0 127.0.0.1:59442         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59380         ESTABLISHED
tcp        0      0 127.0.0.1:59418         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59458         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59206         ESTABLISHED
tcp        0      0 127.0.0.1:59236         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59468         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:59572         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59360         ESTABLISHED
tcp        0      0 127.0.0.1:59358         127.0.0.1:8080          ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59598         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59326         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59626         ESTABLISHED
tcp        0      0 127.0.0.1:8080          127.0.0.1:59298         ESTABLISHED
```

This is how it's being used:
```php
		$application = new pusher();

		$loop = Factory::create();

		$context = new Context( $loop );
		$pull    = $context->getSocket( \ZMQ::SOCKET_PULL );
		$pull->bind( 'tcp://127.0.0.1:5555' );
		$pull->on( 'message', array( $application, 'onAppNotification' ) );

		$webSock   = new Server( '0.0.0.0:8080', $loop );
		$webServer = new IoServer(
			new HttpServer(
				new WsServer(
					new WampServer(
						$application
					)
				)
			),
			$webSock
		);
		
		$loop->run();
```

**Comment by cboden** [comment]
Are you also logging closed connections?

**Comment by ChubbyNinja** [comment]
Those connections are established, so I don't think they are logging closed connections, I use apache2 ProxyPass, so perhaps that isn't closing connections?

I restarted the cli script and i instantly jumped up to 300 connections. Seemed like an odd thing to occur.

**Comment by cboden** [comment]
Ah, I didn't realize those were all so quickly. I was wondering if you had many incoming/outgoing connections but the disconnect simply wasn't being logged. 

I'm unfamiliar with how Apache's ProxyPass works, but if all those are created upon starting it up, I'd guess it's preemptively pooling connections for future connections to use?

**Comment by ChubbyNinja** [comment]
It wasn't doing that originally, just suddenly apache2 was max connections
because the websockets had done this.

I changed port and don't expose the wss client until the user is
authenticated and it seems to be stable now.

I've also improved the authentication I had on the wss end, explicitly
closing connections. I think there is a mix of things that have helped
combat this problem.

The server hosts quite a few busy sites, so I suspect some bot stumbled
across the ws connection and flooded it.

When I randomly look throughout the day, there's are only 10-15 connections
which is expected.

Thanks for looking into this, I think it can be closed for now though!
